kubectl-plugin/src/main.rs

```diff
@@ -34,24 +42,30 @@ async fn main() {
     if let Err(e) = init_rest(cli_args.rest.as_ref()) {
         println!("Failed to initialise the REST client. Error {}", e);
     }
-
-    // Perform the requested operation.
-    let result = match &cli_args.operations {
-        Operations::Get(resource) => match resource {
-            GetResources::Volumes => volume::Volumes::list().await,
-            GetResources::Volume { id } => volume::Volume::get(id).await,
-            GetResources::Pools => pool::Pools::list().await,
-            GetResources::Pool { id } => pool::Pool::get(id).await,
-        },
-        Operations::Scale(resource) => match resource {
-            ScaleResources::Volume { id, replica_count } => {
-                volume::Volume::scale(id, *replica_count).await
-            }
-        },
-    };
-
-    if let Err(e) = result {
-        println!("{}", e);
+    // Check whether the entered format is valid or not
+    if !&cli_args.output.is_empty()
+        && cli_args.output != utils::YAML_FORMAT
+        && cli_args.output != utils::JSON_FORMAT
+    {
+        // Incase of invalid output format, show error and gracefully end.
+        println!("Output not supported for {} format", &cli_args.output);
```

Comment:
You can avoid having to do this by making output a OutputFormat in the structOpt arguments btw.

Reply:
Tried that, but it's not working that way. It needs fmt::Display, to_String and FromStr traits to be implemented. So, this way seemed straightforward.